backend/handler/auth/middleware.py

```diff
@@ -5,17 +5,25 @@
 from joserfc.errors import BadSignatureError
 from joserfc.jwk import OctKey
 from starlette.datastructures import MutableHeaders, Secret
-from starlette.requests import HTTPConnection
+from starlette.requests import HTTPConnection, Request
 from starlette.types import ASGIApp, Message, Receive, Scope, Send
 from starlette_csrf.middleware import CSRFMiddleware
 
 
 class CustomCSRFMiddleware(CSRFMiddleware):
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        # Skip CSRF check if not an HTTP request, like websockets
         if scope["type"] != "http":
             await self.app(scope, receive, send)
             return
 
+        request = Request(scope, receive)
+
+        # Skip CSRF check if Authorization header is present
+        if "Authorization" in request.headers:
```

Comment:
To avoid false positives and requests that skip CSRF without sending a valid header, I would change this to:

```python
auth_scheme = request.headers.get("Authorization", "").split(" ", 1)[0].lower()
if auth_scheme == "bearer" or auth_scheme == "basic":
```